src/common/daft-config/src/lib.rs

```diff
@@ -57,6 +57,8 @@ pub struct DaftExecutionConfig {
     pub enable_aqe: bool,
     pub enable_native_executor: bool,
     pub default_morsel_size: usize,
+    pub enable_pre_shuffle_merge: bool,
```

Comment:
WDYT if instead we just have this as `shuffle_algorithm` and this can be `map-reduce, pre_shuffle_merge, etc`